pkg/chunk/gcp/bigtable_index_client.go

```diff
@@ -41,24 +44,37 @@ type Config struct {
 
 	TableCacheEnabled    bool
 	TableCacheExpiration time.Duration
+
+	WriteLimit        int
+	MaxWriteBurstSize int
 }
 
+var (
+	rateLimitedWriteRequest = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "bigtable_rate_limited_requests_total",
+		Help:      "Total count of rate limited write requests.",
+	}, []string{"operation"})
+)
+
 // RegisterFlags adds the flags required to config this to the given FlagSet
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Project, "bigtable.project", "", "Bigtable project ID.")
 	f.StringVar(&cfg.Instance, "bigtable.instance", "", "Bigtable instance ID.")
 	f.BoolVar(&cfg.TableCacheEnabled, "bigtable.table-cache.enabled", true, "If enabled, once a tables info is fetched, it is cached.")
 	f.DurationVar(&cfg.TableCacheExpiration, "bigtable.table-cache.expiration", 30*time.Minute, "Duration to cache tables before checking again.")
-
+	f.IntVar(&cfg.WriteLimit, "bigtable.write-rate-limit", 0, "BigTable rate limiter for write operations, by default it is disabled.")
+	f.IntVar(&cfg.MaxWriteBurstSize, "bigtable.write-limit-burst", 0, "BigTable maximum burst size for write operations, by default it is disabled.")
```

Comment:
A couple of tiny things to keep more consistency with the rest of the config:
1. `bigtable.write-rate-limit-burst-size`
2. `BigTable maximum burst size for write operations when the rate limit is enabled`
3. Replace `by default it is disabled`, with `0 is disabled` (both CLI flags)

Reply:
done.